### PR TITLE
➕ Add Typescript

### PR DESCRIPTION
**Pull request by saschahaselhuhn**
- Adding typescript
- renaming js files to tsx / ts

https://nextjs.org/docs/basic-features/typescript

### CODE REVIEW

**Review by phlgr — CHANGES_REQUESTED** [review]
Where is your installed module in the package.json and the package-lock.json?

**Review by mewiemers — APPROVED** [review]
only one file is seen here

**Review by M-Sieger — APPROVED** [review]
gj !

**Comment by m-wiemers** [comment]
It seems something is missing 🌎

File: package.json [hunk 1]
```diff
@@ -9,9 +9,12 @@
     "prettify": "prettier --write ."
   },
   "dependencies": {
+    "@types/node": "^14.14.33",
+    "@types/react": "^17.0.3",
     "next": "10.0.8",
     "react": "17.0.1",
-    "react-dom": "17.0.1"
+    "react-dom": "17.0.1",
+    "typescript": "^4.2.3"
```
Comment (lmachens) [inline package.json]: Looks like these should be `devDependencies` and not `dependencies`

**Review by lmachens — CHANGES_REQUESTED** [review]

**Review by lmachens — APPROVED** [review]
🚀

**Review by phlgr — APPROVED** [review]
🚀

**Comment by saschahaselhuhn** [comment]
- Created tsconfig.json with: touch tsconfig.json
- Ran next with: npm run dev
- Installed required packages with: npm install --save-dev typescript @types/react @types/node
- Renamed _app.js and index.js to _app.tsx and index.tsx
- Renamed hello.js to hello.ts
- Configured tsconfig.json with: npm run dev
- Restarted server with npm run dev

https://nextjs.org/docs/basic-features/typescript

**Pull request merged**